packages/nestjs-authorization/src/authorization.module.ts

```diff
@@ -1,10 +1,33 @@
-import { Module } from '@nestjs/common';
+import { DynamicModule, Module } from '@nestjs/common';
 import { PermissionService } from './permission.service';
 import { ConditionService } from './condition.service';
+import { PERMISSION_PROVIDER } from './permission.provider';
+import { AuthorizationModuleOptions } from './authorization-module-options.interface';
 
-@Module({
-  imports: [],
-  providers: [ConditionService, PermissionService],
-  exports: [ConditionService, PermissionService],
-})
-export class AuthorizationModule {}
+@Module({})
+export class AuthorizationModule {
+  private static module: DynamicModule;
+
+  static register({
```

Comment:
what about registerAsync?

Reply:
Why would we need that?

Reply:
E.g. if your permission provider has dependencies. e.g. ConfigService.

Other modules offer something like this:

```typescript
TypeOrmModule.forRootAsync({
      imports: [ConfigModule],
      useFactory: (config: ConfigService) => config.get('database'),
      inject: [ConfigService],
    }),
```